### PR TITLE
Patch 03

### PR DESCRIPTION
**Pull request by PLASMAchicken**
Replace Timer with BPCoreLib Interval
Added /giveammo
Added optional argument to /launch <player> <thrust>
Make use of BP's Test ExecutionMode for OnTryRegister

### CODE REVIEW

File: src/BPEssentials/Events/OnRegister.cs [hunk 1]
```diff
@@ -11,59 +11,19 @@ namespace BPEssentials.RegisteredEvents
...
-            if (!svManager.HandleWhitelist(authData.username))
-            {
-                svManager.RegisterFail(authData.connection, "Account not whitelisted");
+                svManager.RegisterFail(connectionData.connection, $"Your Username can only contain letters: A-Z a-z 0-9 _ -");
                 return false;
             }
-
-            // Don't allow multi-boxing, WebAPI doesn't prevent this
-            foreach (ShPlayer p in EntityCollections.Humans)
+            if (Core.Instance.Settings.General.DisableAccountOverwrite)
```
Comment (UserR00T) [inline src/BPEssentials/Events/OnRegister.cs:22]: Doesn't this block registering each and every time?

File: src/BPEssentials/Events/OnRegister.cs [hunk 2]
```diff
@@ -11,59 +11,19 @@ namespace BPEssentials.RegisteredEvents
...
             {
-                if (p.username == authData.username)
-                {
-                    svManager.RegisterFail(authData.connection, "Account still logged in");
-                    return false;
-                }
+                svManager.RegisterFail(connectionData.connection, "This Name has already been registerd and this Server has disabled overwriting Accounts!");
```
Comment (UserR00T) [inline src/BPEssentials/Events/OnRegister.cs:24]: Minor typo (registerd -> registered)
```suggestion
                svManager.RegisterFail(connectionData.connection, "This name has already been registered and this server has disabled overwriting accounts!");
```

File: src/BPEssentials/Events/OnRegister.cs [hunk 3]
```diff
@@ -11,59 +11,19 @@ namespace BPEssentials.RegisteredEvents
...
-                svManager.AddNewPlayer(connectionData, playerData?.Persistent);
-            }
-        }
-
-        public bool ValidateUser(SvManager svManager, ConnectionData authData)
-        {
-            if (!svManager.HandleWhitelist(authData.username))
-            {
-                svManager.RegisterFail(authData.connection, "Account not whitelisted");
+                svManager.RegisterFail(connectionData.connection, $"Your Username can only contain letters: A-Z a-z 0-9 _ -");
```
Comment (UserR00T) [inline src/BPEssentials/Events/OnRegister.cs:19]: Changed sentence
```suggestion
                svManager.RegisterFail(connectionData.connection, $"Your username can only contain the following characters: A-Z a-z 0-9 _ -");
```

**Review by UserR00T — CHANGES_REQUESTED** [review]
Just have a minor question regarding the `DisableAccountOverwrite` option, and some typo fixes.

File: src/BPEssentials/Events/OnRegister.cs (hunk 1 above)
Comment (PLASMAchicken) [reply src/BPEssentials/Events/OnRegister.cs:22]: Yeah, this was in a nested if.

Comment (UserR00T) [reply src/BPEssentials/Events/OnRegister.cs:22]: Alright, just re-request review when ready.

**Review by UserR00T — APPROVED** [review]
Thanks!

**Pull request merged**